### PR TITLE
Use flake8 type checking

### PR DESCRIPTION
**Pull request by radoering**
# Pull Request Check List

Relates-to: #4776

### CODE REVIEW

File: src/poetry/utils/password_manager.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import logging
 
 from typing import TYPE_CHECKING
-from typing import Dict
-from typing import Optional
+from typing import Dict  # noqa: TC002  # noqa: TC002
```
Comment (neersighted) [inline src/poetry/utils/password_manager.py:4]: Looks like the `# noqa` got added twice here.

**Review by neersighted — CHANGES_REQUESTED** [review]
Awesome! I like this a lot more than I even thought I would -- it really makes the usage of imports unambiguous.

A few corrections/nits to pick, and then also one question -- since upstream seems open to adding a whitelist, is it worth adding those so we can drop `# noqa`? The good news is when the whitelist is added, yesqa should automate removing those comments.

Alternatively, there is https://pypi.org/project/future-annotations/, but I'm not sure how it will interact with the rest of our tooling.

File: src/poetry/masonry/builders/__init__.py [hunk 2]
```diff
@@ -1 +1 @@
-from poetry.masonry.builders.editable import EditableBuilder
+from poetry.masonry.builders.editable import EditableBuilder  # noqa: TC002
```
[the remark this answers is not in the document]
Comment (radoering) [reply src/poetry/masonry/builders/__init__.py:1]: Not sure. Probably, suitable for most `__init__.py` files. However, some files like `poetry/mixology/__init__.py` should probably not be excluded. Maybe, the same applies to `F401`. 🤷‍♂️ I'll change it if you like.

**Comment by radoering** [comment]
> A few corrections/nits to pick, and then also one question -- since upstream seems open to adding a whitelist, is it worth adding those so we can drop `# noqa`? The good news is when the whitelist is added, yesqa should automate removing those comments.

I'll try the whitelist as soon as it has been added upstream. That should remove a lot of `# noqa`.

**Review by finswimmer — APPROVED** [review]
Beautiful :heart_eyes:

**Pull request merged**